scripts/host-check

```diff
@@ -790,65 +792,114 @@ def check_hugepages() -> CheckFuncReturn:
+                f"Loaded PCI drivers: {supported_str}.\n"
+                "Run 'modprobe <pci driver>' to load a driver.",
+            )
+        else:
+            return CheckState.SUCCESS, f"Loaded PCI drivers: {supported_str}"
     else:
-        return (
-            CheckState.FAILED,
-            "The kernel module vfio-pci is not installed. It may be\n"
-            "possible to install using your distro's package manager.",
-        )
+        if installed_not_supported:
+            return (
+                CheckState.FAILED,
+                "None of the expected PCI drivers are loaded.\n"
+                f"The following PCI drivers are installed but not loaded: {installed_not_supported_str}.\n"
+                "Run 'modprobe <pci driver>' to load a driver.",
+            )
+        else:
+            return (
+                CheckState.FAILED,
+                "No PCI drivers are loaded or installed.\n"
+                "Must have either the vfio-pci or igb_uio kernel module loaded.\n"
+                "It may be possible to install using your distro's package manager.",
+            )
 
 
 def check_iommu() -> CheckFuncReturn:
     """Check IOMMU is set up correctly for the vfio-pci kernel module."""
+    if not PCIDriver("vfio-pci").is_supported():
+        return CheckState.NEUTRAL, "vfio-pci driver unavailable"
+
+    # If igb_uio is supported, treat failures and warnings as info, because
+    # even though IOMMU isn't enabled, igb_uio can be used.
+    fail_state = CheckState.FAILED
+    warning_state = CheckState.WARNING
+    if PCIDriver("igb_uio").is_supported():
+        fail_state = CheckState.NEUTRAL
+        warning_state = CheckState.NEUTRAL
+
     recommendation = "IOMMU is recommended for security when using the vfio-pci kernel driver."
     noiommu_filepath = "/sys/module/vfio/parameters/enable_unsafe_noiommu_mode"
     iommu_dev_filepath = "/sys/class/iommu/*/devices/*"
+
     try:
         with open(noiommu_filepath, "r", encoding="utf-8") as f:
             no_iommu_mode = f.read().strip().upper() in ("Y", "1")
             if no_iommu_mode:
                 return (
-                    CheckState.WARNING,
+                    warning_state,
                     "vfio-pci is set up in no-IOMMU mode, but IOMMU is recommended for security.",
                 )
     except OSError:
-        return (
-            CheckState.WARNING,
-            "Failed to check whether vfio-pci 'no-IOMMU' mode is enabled by reading\n"
-            f"{noiommu_filepath}.\n" + recommendation,
-        )
+        # If the no-IOMMU file does not exist, then no-IOMMU mode can't be enabled.
+        pass
+
     # Check if IOMMU is enabled on the host
     try:
         iommu_devices = [
             os.path.basename(p) for p in glob.glob(iommu_dev_filepath)
         ]
         if not iommu_devices:  # check if the directory is empty
             return (
-                CheckState.FAILED,
+                fail_state,
```

Comment:
I think this should be a warning (`warning_state`) - if we're allowing igb_uio (which doesn't support IOMMU) then it probably doesn't make sense to be so strict for vfio-pci.